modules/dataframe/src/processors/filelist.cpp

```diff
@@ -0,0 +1,186 @@
+    , bnlInport_{"bnlInport"}
+    , bnlOutport_{"bnlOutport"}
+    , directory_{"directory", "Directory"}
+    , refresh_{"refresh", "Refresh"}
+    , filter_{"filter", "Filter",
+              "Set a regexp filter to match filenames to include in the list"_help}
+    , selectedIndex_{"selectedIndex", "Selected Index", util::ordinalCount(size_t{0}, size_t{100})}
+    , highlightIndex_{"highlightIndex", "Highlight Index",
+                      util::ordinalCount(size_t{0}, size_t{100})}
+    , selected_{"selected", "Selected"}
+    , highlight_{"highlight", "Highlight"}
+    , cycleFiles_{"cycleFiles", "Cycle throw all files",
+                  "This will each files a the selected one and wait for the network to evaluate "
+                  "before setting the next one"_help,
+                  [this]() { cycleFiles(); }} {
+
+    addPorts(outport_, bnlInport_, bnlOutport_);
+    addProperties(directory_, refresh_, filter_, selectedIndex_, selected_, highlightIndex_,
+                  highlight_, cycleFiles_);
+
+    selected_.setReadOnly(true);
+    highlight_.setReadOnly(true);
+
+    bnlOutport_.getManager().setParent(&bnlInport_.getManager());
+    bnlOutport_.getManager().onBrush([this](BrushingAction action, const BrushingTarget& target,
+                                            const BitSet& indices, std::string_view source) {
+        if (action == BrushingAction::Select && target == BrushingTarget::Row) {
+            if (files_.empty() || indices.empty()) {
+                selected_.set("");
+            } else {
+                const Property::OnChangeBlocker block{selectedIndex_};
+                selectedIndex_.set(indices.min());
+                selected_.set(files_[std::min(files_.size() - 1, size_t{indices.min()})].path());
+            }
+        } else if (action == BrushingAction::Highlight && target == BrushingTarget::Row) {
+            if (files_.empty() || indices.empty()) {
+                highlight_.set("");
+            } else {
+                const Property::OnChangeBlocker block{highlightIndex_};
+                highlightIndex_.set(indices.min());
+                highlight_.set(files_[std::min(files_.size() - 1, size_t{indices.min()})].path());
+            }
+        }
+    });
+
+    selectedIndex_.onChange([this]() {
+        bnlOutport_.getManager().brush(BrushingAction::Select, BrushingTarget::Row,
+                                       {selectedIndex_.get()});
+    });
+    highlightIndex_.onChange([this]() {
+        bnlOutport_.getManager().brush(BrushingAction::Highlight, BrushingTarget::Row,
+                                       {highlightIndex_.get()});
+    });
+}
+
+void FileList::cycleFiles() {
+    auto* net = getNetwork();
+    auto* app = net->getApplication();
+
+    app->dispatchFrontAndForget([app, net, this]() {
+        for (size_t i = 0; i < files_.size(); ++i) {
+            log::info("Loading: {} {:?g}", i, files_[i].path());
+            selectedIndex_.set(i);
```

Comment:
no option to cancel the iteration? For example by triggering the button again.

Reply:
puh